advisor/src/test/kotlin/advisors/VulnerableCodeTest.kt

```diff
@@ -77,130 +77,102 @@ class VulnerableCodeTest : WordSpec({
-                    .willReturn(
-                        aResponse().withStatus(200)
-                            .withBodyFile("response_packages.json")
-                    )
-            )
-
-            stubVulnerability("v1", "CVE-2014-8242", 42.0f)
-            stubVulnerability("v2", "CVE-2009-1382", 11.0f)
-            stubFor(
-                get(urlPathEqualTo("/api/vulnerabilities/v3"))
+                post(urlPathEqualTo("/api/packages/bulk_search/"))
                     .willReturn(
-                        aResponse().withStatus(200)
-                            .withBodyFile("response_vulnerability_incomplete.json")
+                        aResponse().withStatus(500)
                     )
             )
 
-            val vulnerableCode = createVulnerableCode(wiremock)
-            val packagesToAdvise = resultFile().readValue<OrtResult>().getPackages(false).map { it.pkg }
-
-            vulnerableCode.retrievePackageVulnerabilities(packagesToAdvise).mapKeys { it.key.id } shouldNotBeNull {
-                val strutsResults = getValue(idStruts)
-                val expStrutsVulnerabilities = listOf(
-                    Vulnerability(
-                        id = "CVE-2009-1382",
-                        createReferences(URI("http://localhost:8000/api/vulnerabilities/v2/"), severity = 11.0f)
-                    ),
-                    Vulnerability(
-                        id = "",
-                        createReferences(URI("http://localhost:8000/api/vulnerabilities/v3/"), severity = -1f)
-                    )
-                )
-                strutsResults.flatMap { it.vulnerabilities } should containExactlyInAnyOrder(expStrutsVulnerabilities)
-            }
+            expectErrorResult(wiremock)
         }
 
-        "return failures if a vulnerability cannot be resolved" {
-            stubFor(
-                post(urlPathEqualTo("/api/packages/bulk_search/"))
-                    .withRequestBody(equalToJson(packagesRequestJson, false, false))
-                    .withHeader("Content-Type", equalTo("application/json; charset=UTF-8"))
-                    .willReturn(
-                        aResponse().withStatus(200)
-                            .withBodyFile("response_packages.json")
-                    )
-            )
+        "filter out packages without vulnerabilities" {
+            wiremock.stubPackagesRequest("response_packages_no_vulnerabilities.json")
+            val vulnerableCode = createVulnerableCode(wiremock)
+            val packagesToAdvise = inputPackages()
 
-            stubVulnerability("v1", "CVE-2014-8242", 42.0f)
-            stubVulnerability("v2", "CVE-2009-1382", 11.0f)
-            stubVulnerability("v3", "CVE-2019-CoV19", 77.0f, statusCode = 500)
+            val result = vulnerableCode.retrievePackageVulnerabilities(packagesToAdvise).mapKeys { it.key.id }
 
-            expectErrorResult(wiremock)
+            result.keys should containExactly(idStruts)
```

Comment:
Why no `containExactlyInAnyOrder` here as below?

Reply:
As there is only a single element, I found it moot to state that the order is irrelevant.